### PR TITLE
Document runbook attachment rule and step rule

### PR DESCRIPTION
**Pull request by nearbycoder**
Description
This PR documents the attachment_rule and step rule properties for runbooks.

Testing plan
The goal of this testing plan is to make sure the documentation makes sense and is as accurate as it can be. While going through this test plan, please check the documentation and configs against what you see in the UI and what you see at GET https://api.firehydrant.io/v1/runbooks/actions?per_page=100

Run the docs through the doc preview tool to make sure they look okay
https://registry.terraform.io/tools/doc-preview

Read through the docs looking for typos or inconsistencies.

### CODE REVIEW

File: docs/guides/runbooks_attachment_rule.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+
+## Types of conditions
+
+We derive the types of conditions that can be set based on our API
+```
+https://api.firehydrant.io/v1/fh-attributes/data_bags/system-runbook-attachment-attributes
+```
+
+With the example payload below you can see that we have a list of attributes that can be selected and we can see the `opcode` or operators are able to be used for these attributes. 
+Give these operators we can infer that we can either have a condition where the slack channel does or does not exist.
```
Comment (lafentres) [inline docs/guides/runbooks_attachment_rule.md:111]:
```suggestion
In the example payload below you can see that we have a list of attributes that can be selected and we can see the `opcode` or operators that can be used for these attributes. 
Given these operators, we can infer that we can either have a condition where the Slack channel does or does not exist.
```

File: docs/guides/runbooks_attachment_rule.md [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+      ]
+    }
+  }
+}
+
+```
+
+## Value based conditions
+
+Operators of conditions can be assigned values that are from data saved within different parts of our system. For example the below data we get back from the `system-runbook-attachment-attributes` api shows that we can select multiple operators that take in an `Array` of `IncidentRole` data values. Looking into the `IncidentRole` type we also see values that specify an `async` url that we can use to check and grab the data necessary to fill in the `type`, `value`, and `label` fields required for the user_data.
```
Comment (lafentres) [inline docs/guides/runbooks_attachment_rule.md:143]:
```suggestion
Operators of conditions can be assigned values that come from data saved in different parts of our system. For example, the below data we get back from the runbook attachment attributes API shows that we can select multiple operators that take in an array of `IncidentRole` data values. Looking at the `IncidentRole` type, we also see values that specify an `async` url that we can use to get the data necessary to fill in the `type`, `value`, and `label` fields required for the `user_data` attribute.
```

File: docs/guides/runbooks_attachment_rule.md [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+        },
+        {
+          type = "IncidentRole",
+          value = "44e8b65d-7682-4279-85f0-ed68cba052cc",
+          label = "Communication"
+        }
+      ],
+    }
+  }
+})
```
Comment (lafentres) [inline docs/guides/runbooks_attachment_rule.md:231]:
```suggestion
attachment_rule = jsonencode({
  logic = {
    includes_any = [
      {
        var = "incident_assigned_roles"
      },
      {
        var = "usr.1"
      }
    ]
  }
  user_data = {
    "1" = {
      type = "Array[IncidentRole]"
      value = [
        {
          type = "IncidentRole"
          value = data.firehydrant_incident_role.commander.id
          label = data.firehydrant_incident_role.commander.name
        },
        {
          type = "IncidentRole"
          value = data.firehydrant_incident_role.communications.id
          label = data.firehydrant_incident_role.communications.name
        }
      ]
      label = null
    }
  }
})
```

File: docs/guides/runbooks_attachment_rule.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+    ]
+  },
+  user_data = {
+    "1" = {
+      type  = "Milestone",
+      value = "resolved",
+      label = "Resolved"
+    }
+  }
+})
```
Comment (lafentres) [inline docs/guides/runbooks_attachment_rule.md:256]:
```suggestion
attachment_rule = jsonencode({
  logic = {
    eq = [
      {
        var = "incident_current_milestone"
      },
      {
        var = "usr.1"
      }
    ]
  }
  user_data = {
    "1" = {
      type  = "Milestone"
      value = "resolved"
      label = "Resolved"
    }
  }
})
```

**Review by lafentres — COMMENTED** [review]
just some minor changes for clarity and some fixes for small bugs i found when trying out the attachment rules

**Review by nearbycoder — COMMENTED** [review]
I can't approve but overall the changes LGTM! This is our first step to actually having people outside of FH understand what our conditions logic looks, like. Next would be porting this over to the API documentation.

**Review by lafentres — APPROVED** [review]

**Pull request merged**